titus-server-master/src/main/java/com/netflix/titus/master/scheduler/constraint/IpAllocationConstraint.java

```diff
@@ -98,4 +99,18 @@ public Result evaluate(TaskRequest taskRequest, VirtualMachineCurrentState targe
                 })
                 .orElse(INVALID_IP_ALLOCATION_ZONE);
     }
+
+    public static boolean isInUseIpAllocationConstraintReason(String reason) {
+        return reason != null && reason.startsWith(IP_ALLOCATION_ALREADY_IN_USE_REASON_PREFIX);
+    }
+
+    public static Optional<String> getTaskIdFromIpAllocationInUseReason(String reason) {
+        return reason.startsWith(IP_ALLOCATION_ALREADY_IN_USE_REASON_PREFIX)
+                ? Optional.of(reason.substring(IP_ALLOCATION_ALREADY_IN_USE_REASON_PREFIX.length()))
+                : Optional.empty();
+    }
+
+    private static Result getIpAllocationInUseResult(String taskId) {
```

Comment:
How often do you expect this result to be generated? A major issue we had in the past was creating too many `Result` objects leading to slowdowns caused by GC which is why we moved to static results for every constraint even though we lost visibility.

Reply:
Not often. I don't expect wide spread use of this feature to begin with and then this catches usages where either the user did something wrong or is part of a rolling deploy (e.g., redeploying ZK).

I understand the GC issue, though also don't have an obviously better place to stash the ID of the blocking task.